cpp/src/bindings/bindings_linear_animation_instance.cpp

```diff
@@ -64,10 +60,10 @@ extern "C"
                 enumField = ::pingPongLoopField;
                 break;
             }
+            
+            loopValue = env->GetStaticObjectField(::loopClass, enumField);
         }
 
-        jobject loopValue = env->GetStaticObjectField(::loopClass, enumField);
-
         return loopValue;
```

Comment:
Here the goal is to extract whether an animation has looped or not.
advance() returns a bool to notify if it advanced, but there wasn't a clear way to know if the animation had looped around, or it pingpong'd. 
So when a loop happens, the flag that is passed in as a parameter gets raised, this is why the code queries for the loop type and reconciles it with the Koktlin enum.

This is indeed living also in the cpp side, as there's a PR pending over there!

As of now, I needed the flag on the Android side to build the example, but this should indeed return the same value as advance. 
We just need to set this up so that we can have both values returned in Kotlin too (a field to be set on the animation instance? An object that wraps both values?)

Reply:
Ideally, if we want to have a 1:1 match between the cpp & the binding, we'd have a way of letting the Kotlin layer of both things, the advance return value, and the didLoop value.
If you think switching for the enum on the Kotlin side is cleaner, I can set it up that way!

Reply:
The idea was to keep the advance() path as lean as possible since this is being called every frame, and returning an object from a function in cpp might lead to unexpected results.

For example, I had put together a completely different solution on the rive-cpp PR, but passing an output parameter here was found to be the solution with the most minimal impact code-wise. So probably it's best to ask @luigi-rosso directly and see what he thinks about returning the object there.